### PR TITLE
db separation & service ctx passed also to rest activator 

### PR DESCRIPTION
**Pull request by tworec**
Market, Activity and Payment services has their own db files from now on.
This also enables access to DB from both GSB and rest activators.

As a side kick it also contains `ya-persistance` cleanup: migrations, models and schema is moved to `ya-activity`

### CODE REVIEW

File: core/serv/src/main.rs [hunk 1]
```diff
@@ -136,35 +141,56 @@ impl TryFrom<&CliArgs> for CliCtx {
     }
 }
 
+#[derive(Clone)]
 struct ServiceContext {
-    db: DbExecutor,
-    data_dir: PathBuf,
```
Comment (nieznanysprawiciel) [inline core/serv/src/main.rs:141]: This version of decentralized market still has data_dir. It shouldn't compile

File: core/serv/src/main.rs [hunk 2]
```diff
@@ -136,35 +141,56 @@ impl TryFrom<&CliArgs> for CliCtx {
...
-    fn component(&self) -> PathBuf {
-        self.data_dir.clone()
+impl ServiceContext {
+    fn make_entry<S: 'static>(path: &PathBuf, name: &str) -> Result<(TypeId, DbExecutor)> {
+        Ok((TypeId::of::<S>(), DbExecutor::from_data_dir(path, name)?))
+    }
+
+    fn from_data_dir(path: &PathBuf, name: &str) -> Result<Self> {
+        let default_db = DbExecutor::from_data_dir(path, name)?;
+        default_db.apply_migration(ya_persistence::migrations::run_with_output)?;
```
Comment (nieznanysprawiciel) [inline core/serv/src/main.rs:168]: This line will add activity migrations to all databases. In my code there were no migrations done here excluding default database. Each service runs migrations by itself.

Comment (nieznanysprawiciel) [reply core/serv/src/main.rs:168]: Maybe a little more explanation: activity database uses migrations defined in ya-persistence. But all other services have their own migrations sets.

Comment (tworec) [reply core/serv/src/main.rs:168]: I'll move `ya-persistence::migrations` to `ya-activity`

Comment (nieznanysprawiciel) [reply core/serv/src/main.rs:168]: It's probably good choice

File: core/serv/src/main.rs (hunk 1 above)
Comment (tworec) [reply core/serv/src/main.rs:141]: done

**Comment by tworec** [comment]
@prekucki @nieznanysprawiciel @mfranciszkiewicz pls approve

**Pull request merged**